File: pdu/python/hako_srv_msgs/pdu_cdr_conv_AddTwoIntsResponse.py
from .pdu_pytype_AddTwoIntsResponse import AddTwoIntsResponse
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_AddTwoIntsResponse(writer: CdrWriter, src: AddTwoIntsResponse):
    writer.write_int64(src.sum)


def cdr_body_to_py_AddTwoIntsResponse(reader: CdrReader, dst: AddTwoIntsResponse):
    dst.sum = reader.read_int64()
    return dst


def py_to_cdr_AddTwoIntsResponse(src: AddTwoIntsResponse) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_AddTwoIntsResponse(writer, src)
    return writer.bytes()


def cdr_to_py_AddTwoIntsResponse(cdr_payload) -> AddTwoIntsResponse:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = AddTwoIntsResponse()
    return cdr_body_to_py_AddTwoIntsResponse(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_AddTwoIntsResponse.js
import { AddTwoIntsResponse } from './pdu_jstype_AddTwoIntsResponse.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduAddTwoIntsResponseConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { AddTwoIntsResponse } src
     */
    static to_cdr_body(writer, src) {
        writer.write_int64(src.sum);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { AddTwoIntsResponse } dst
     * @returns { AddTwoIntsResponse }
     */
    static cdr_body_to_js(reader, dst) {
        dst.sum = reader.read_int64();
        return dst;
    }

    /**
     * @param { AddTwoIntsResponse } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { AddTwoIntsResponse }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new AddTwoIntsResponse());
    }
}
